bleak/backends/bluezdbus/client.py

```diff
@@ -437,7 +437,9 @@ async def get_services(self) -> BleakGATTServiceCollection:
             self.services.add_characteristic(
                 BleakGATTCharacteristicBlueZDBus(char, object_path, _service[0].uuid)
             )
-            self._char_path_to_handle[object_path] = char.get("Handle")
+
+            # D-Bus object path contains handle as last 4 characters of 'charYYYY'
+            self._char_path_to_handle[object_path] = int(object_path[-4:], 16)
```

Comment:
This is the actual fix for the problem then. But I agree that this is more concise and plain to read than the regex.